compiler/resolution/visibleFunctions.cpp

```diff
@@ -205,14 +195,61 @@ static BlockStmt* getVisibleFunctions(const char*           name,
   // avoid infinite recursion due to modules with mutual uses
   //
   if (visited.find(block) == visited.end()) {
-    bool canSkipThisBlock = true;
 
-    if (isModuleSymbol(block->parentSymbol)) {
+    bool moduleBlock = false;
+    bool fnBlock = false;
+    ModuleSymbol* inMod = block->getModule();
+    FnSymbol* inFn = block->getFunction();
+    BlockStmt* instantiationPt = NULL;
+    if (block->parentExpr != NULL) {
+      // not a module or function level block
+    } else if (inMod && block == inMod->block) {
+      moduleBlock = true;
+    } else if (inFn != NULL) {
+      // TODO - probably remove this assert
+      INT_ASSERT(block->parentSymbol == inFn ||
+                 isArgSymbol(block->parentSymbol) ||
+                 isShadowVarSymbol(block->parentSymbol));
+      // Could also check
+      fnBlock = true;
+      if (inFn->instantiationPoint && inFn->instantiationPoint->parentSymbol)
+        instantiationPt = inFn->instantiationPoint;
+    }
+
+    if (call->id == breakOnResolveID) {
+      if (moduleBlock)
+        printf("visible fns: block %i module %s %s:%i\n",
+               block->id, inMod->name,
+               block->fname(), block->linenum());
+      else if (fnBlock)
+        printf("visible fns: block %i fn %s %s:%i\n",
+               block->id, inFn->name,
+               block->fname(), block->linenum());
+      else
+        printf("visible fns: block %i %s:%i\n",
+               block->id,
+               block->fname(), block->linenum());
+
+      if (instantiationPt) {
+        printf("  instantiated from block %i %s:%i\n",
+               instantiationPt->id,
+               instantiationPt->fname(), instantiationPt->linenum());
+      }
+    }
+
+    /*if (moduleBlock || fnBlock)  -- see comment below*/
+    {
+
+      // e.g. in associative.chpl primer, instatiation occurs in a
+      // block that isn't a fn or module block.
+      // But, we could probably optimize by knowing:
+      //   - which blocks are irrelevant
+      //   - which blocks are traversed as outer blocks
       visited.insert(block);
     }
 
     if (VisibleFunctionBlock* vfb = visibleFunctionMap.get(block)) {
-      canSkipThisBlock = false; // cannot skip if this block defines functions
+      // note, defines functions
```

Comment:
Should it be colon, not comma, in the above comment ?
Or maybe convey the point differently?
Ditto a similar comment you inserted below.